### PR TITLE
Rework controller and autorization flow

### LINKED ISSUE #164: With configuration.xml set to guest=no, login is impossible
In my testing of @line-o's https://github.com/eXist-db/eXide/pull/142, setting `guest=no` prevents users from even getting to the page where they would log in.  I propose reverting this PR and restoring the ability of users to access the log in page so that they can authenticate.

### PR DESCRIPTION
**Pull request by line-o**
Fixes #164

- add login page as static HTML (adapted from dashboard)
- rewrite controller to always authenticate requests
- add routes handling new flow
- add background SVG for login page

Here is a screenshot of Safari (top left), Edge (top right), Chrome (bottom left), Firefox (bottom right) showing the login page.

Screenshot 2023-02-13 at 23 36 06

### CODE REVIEW

**Comment by line-o** [comment]
TIL that `git push -d` deletes a branch :/ Well, I recreated the PR #594

**Review by duncdrum — APPROVED** [review]
looks great thank @line-o

File: controller.xq [hunk 1]
```diff
@@ -1,128 +1,159 @@
...
+(: public :)
+
 if ($exist:path eq '') then
     <dispatch xmlns="http://exist.sourceforge.net/NS/exist">
-        <redirect url="{request:get-uri()}/"/>
+        <redirect url="{$local:uri}/"/>
     </dispatch>
 
-else if ($exist:path eq '/') then
+else if ($exist:path eq '/' and $user-allowed) then
```
Comment (line-o) [inline controller.xq]: `and $user-allowed` is not needed here (unauthenticated users will have an additional redirect)

File: controller.xq [hunk 2]
```diff
@@ -1,128 +1,159 @@
...
-            else
-                (
-                    response:set-status-code(401),
-                    <status>fail</status>
-                )
-        } catch * {
-            response:set-status-code(401),
-            <status>{$err:description}</status>
-        }
+else if (
```
Comment (line-o) [inline controller.xq]: This route can be moved below handling of unauthorised requests

File: login.html [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        .submit-button:-moz-focusring { outline: 1px dotted ButtonText }
+
+    </style>
+</head>
+<body class="login-page">
+    <main class="login-box">
+        <h1 class="header">Sign in</h1>
+        <form action="login" method="post">
+            <input type="hidden" name="duration" value="P7D"/>
+            <input id="route" type="hidden" name="route"/>
```
Comment (line-o) [inline login.html]: this input is not needed here

**Comment by joewiz** [comment]
I have checked out this branch and built and installed it into eXist 6.2.0. I can confirm that setting guest=no in eXide's configuration.xml file to cause any request to eXide to redirect to http://localhost:8080/exist/apps/eXide/login.html - with the page appearing as shown in the screenshots above. When guest=yes (as it is by default), this PR does not alter eXide's login UI at all. 

@line-o Is this intentional, or was the new login screen meant to replace the original login UI too?

One suggestion. The placeholder for the user input field `itsme` is unusual. I'd recommend changing this placeholder text to "Username..." Similarly, the placeholder for the password input ifeld `··········` makes it appear as if a password has *already* been entered, so I'd recommend changing this placeholder text to "Password..."

**Comment by line-o** [comment]
@joewiz thank you for taking the time to review my PR.

> Does the login page replace the current login form?

Yes, when guest access is restricted it is unsafe to even show the eXide page. If you are logged in and this can see eXide the form will still work and allow to switch to a different user.

Nothing changes for instances where guest access is allowed.

> unconventional placeholders

I will change the placeholders as you suggested.

**Review by joewiz — APPROVED** [review]

**Pull request merged**